src/mpyl/steps/run_properties.py

```diff
@@ -0,0 +1,77 @@
+"""Module to initiate run properties"""
+import logging
+from pathlib import Path
+from typing import Optional
+
+from ..cli import MpylCliParameters
+from ..project import load_project, Stage, Project
+from ..stages.discovery import find_build_set
+from ..steps.models import RunProperties
+from ..utilities.repo import Repository, RepoConfig
+
+
+def initiate_run_properties(
+    config: dict,
+    properties: dict,
+    cli_parameters: MpylCliParameters = MpylCliParameters(),
+    run_plan: Optional[dict[Stage, set[Project]]] = None,
+    all_projects: Optional[set[Project]] = None,
+) -> RunProperties:
+    if all_projects is None or run_plan is None:
+        with Repository(RepoConfig.from_config(config)) as repo:
+            if all_projects is None:
+                project_paths = repo.find_projects()
+                all_projects = set(
+                    map(
+                        lambda p: load_project(
+                            root_dir=Path(""),
+                            project_path=Path(p),
+                            strict=False,
+                            log=True,
+                            safe=True,
+                        ),
+                        project_paths,
+                    )
+                )
+
+            if run_plan is None:
+                stages = [
+                    Stage(stage["name"], stage["icon"])
+                    for stage in properties["stages"]
+                ]
+                run_plan = find_build_set(
+                    logger=logging.getLogger("mpyl"),
+                    all_projects=all_projects,
+                    changes_in_branch=(
+                        repo.changes_in_branch_including_local()
+                        if cli_parameters.local
+                        else (
+                            repo.changes_in_tagged_commit(cli_parameters.tag)
+                            if cli_parameters.tag
+                            else repo.changes_in_branch()
+                        )
+                    ),
+                    stages=stages,
+                    build_all=cli_parameters.all,
+                    selected_stage=cli_parameters.stage,
+                    selected_projects=cli_parameters.projects,
+                )
+
+    if cli_parameters.local:
+        return RunProperties.for_local_run(
```

Comment:
Do we still need this?

Reply:
We're definitely closer to not needing it anymore, but I think there is still some extra work if you want to get rid of it. Maybe we can make a ticket? I think we should get this bugfix out and I think the current solution is pretty clean already since the function is only called here and there isn't a lot of extra logic around it anymore.